packages/solidstart/src/vite/types.ts

```diff
@@ -0,0 +1,94 @@
+export type SourceMapsOptions = {
+  /**
+   * If this flag is `true`, and an auth token is detected, the Sentry SDK will
+   * automatically generate and upload source maps to Sentry during a production build.
+   *
+   * @default true
+   */
+  enabled?: boolean;
+
+  /**
+   * The auth token to use when uploading source maps to Sentry.
+   *
+   * Instead of specifying this option, you can also set the `SENTRY_AUTH_TOKEN` environment variable.
+   *
+   * To create an auth token, follow this guide:
+   * @see https://docs.sentry.io/product/accounts/auth-tokens/#organization-auth-tokens
+   */
+  authToken?: string;
+
+  /**
+   * The organization slug of your Sentry organization.
+   * Instead of specifying this option, you can also set the `SENTRY_ORG` environment variable.
+   */
+  org?: string;
+
+  /**
+   * The project slug of your Sentry project.
+   * Instead of specifying this option, you can also set the `SENTRY_PROJECT` environment variable.
+   */
+  project?: string;
+
+  /**
+   * If this flag is `true`, the Sentry plugin will collect some telemetry data and send it to Sentry.
+   * It will not collect any sensitive or user-specific data.
+   *
+   * @default true
+   */
+  telemetry?: boolean;
+
+  /**
+   * Options related to sourcemaps
+   */
+  sourcemaps?: {
+    /**
+     * A glob or an array of globs that specify the build artifacts and source maps that will be uploaded to Sentry.
+     *
+     * The globbing patterns must follow the implementation of the `glob` package.
+     * @see https://www.npmjs.com/package/glob#glob-primer
+     */
+    assets?: string | Array<string>;
+
+    /**
+     * A glob or an array of globs that specifies which build artifacts should not be uploaded to Sentry.
+     *
+     * @default [] - By default no files are ignored. Thus, all files matching the `assets` glob
+     * or the default value for `assets` are uploaded.
+     *
+     * The globbing patterns follow the implementation of the glob package. (https://www.npmjs.com/package/glob)
+     */
+    ignore?: string | Array<string>;
+
+    /**
+     * A glob or an array of globs that specifies the build artifacts that should be deleted after the artifact
+     * upload to Sentry has been completed.
+     *
+     * @default [] - By default no files are deleted.
+     *
+     * The globbing patterns follow the implementation of the glob package. (https://www.npmjs.com/package/glob)
+     */
+    filesToDeleteAfterUpload?: string | Array<string>;
+  };
+
+  /**
+   * Enable debug functionality of the SDK during build-time.
+   * Enabling this will give you logs about source maps.
+   */
+  debug?: boolean;
+};
+
+/**
+ *  Build options for the Sentry module. These options are used during build-time by the Sentry SDK.
+ */
+export type SentrySolidStartPluginOptions = {
```

Comment:
m: I think the structure of this object can be improved. You may want things like `org`, `project`, `authToken` top level, as they are not necessarily tied to source maps but to release creation.

Reply:
I aligned this with the [Nuxt](https://github.com/getsentry/sentry-javascript/blob/develop/packages/nuxt/src/common/types.ts#L8-L102) and [Sveltekit](https://github.com/getsentry/sentry-javascript/blob/develop/packages/sveltekit/src/vite/types.ts#L14-L122) package, can change but I think it's better to keep the consistency, wdyt?

Reply:
Similarly to the debug, I decided to keep it like this because this plugin could be used for more than just source maps/sentry's vite plugin in the future. 

The keys would have to be passed along anyway(unless every plugin has access to the entire plugin config) ending up with repeated keys. For the sake of keeping API surface low I think we should stick with this for now and reevaluate if they do need these keys hoisted when the time comes.

Reply:
@lforst I thought about this some more and I think you're right, I flattened this.